### PR TITLE
Feature/toml

### PR DESCRIPTION
**Pull request by sam1el**
Remove dependency on external `toml-rb` gem. 

### Description
The current iteration of the cookbook relies on the `toml-rb` gem which has a few other dependencies of it's own. In support of moving all of this cookbook's functionality to chef-client native as well as allowing ease of use for air-gapped infrastructure, we decided to build this in as a helper library. This also led us to add some more testing to ensure all things are still working as expected.  

### Check List

- [x] All tests pass. See <https://github.com/chef-cookbooks/community_cookbook_documentation/blob/master/TESTING.MD>
- [x] New functionality includes testing.
- [x] New functionality has been documented in the README if applicable
- [x] All commits have been signed for the Developer Certificate of Origin. See <https://github.com/chef-cookbooks/community_cookbook_documentation/blob/master/CONTRIBUTING.MD>

### CODE REVIEW

File: libraries/habitat_toml.rb [hunk 1]
```diff
@@ -0,0 +1,116 @@
...
+            args = sort_pairs(child) << aux_prefix
+
+            dump_pairs(*args)
+          end
+        end
+      end
+
+      def print_prefix(prefix, extra_brackets = false)
+        new_prefix = prefix.join('.')
+        new_prefix = '[' + new_prefix + ']' if extra_brackets
```
Comment (Xorima) [inline libraries/habitat_toml.rb:84]: I think using string interpolation will make this more readable
```suggestion
        new_prefix = "[#{new_prefix}]" if extra_brackets
```

Comment (Xorima) [reply libraries/habitat_toml.rb:84]: Also should the arg be called `extra_brackers` or should it be called `array` as that is what it means?

Comment (sam1el) [reply libraries/habitat_toml.rb:84]: Will work on this now and get it tested locally

Comment (sam1el) [reply libraries/habitat_toml.rb:84]: if that's the best way to do it for clarity, it makes sense to me. do you think we should follow the same pattern you suggested above for line 86?
```suggestion
          @toml_str += '[' + new_prefix + "]\n"
          @toml_str += "[#{new_prefix}]\n"
```

Comment (sam1el) [reply libraries/habitat_toml.rb]: this was tested and implemented. it does make more sense to rename extra_brackets for clarity as well as using string interpolation in a uniform manner

**Pull request merged**